### DISCUSSION TITLE
Shopify Payments Integration

### DISCUSSION THREAD

**Discussion by georgeartem** [veryengineer/collab#23] [General]
Hello there, 

It looks like most of the discussion is around nodes, just pinging to see if anyone in the group might have interest in working on a dogecoin payment gateway for the Shopify ecosystem to help with the adoption side. 

I have a private repo with what my team has been building that I would be happy to extend an invite to. It's been years since I touched any code so any contributions/recommendations would be greatly appreciated. It's using PHP Laravel :-)

**Comment by DogestMaximus** [comment]
I may know just the person ;)

**Comment by qlpqlp** [comment]
Im interested in joining and helping on whatever is needed 🙂❤️

I also started last year the Shopify integration but I stopped wen I realised the requirements of Shopify to be able to list my plugin on their platform 😅

**Reply by georgeartem** [reply]
@qlpqlp I have some private repos. Sent you an invite. Feel free to explore and provide feedback <3

**Comment by patricklodder** [comment]
I tried this in 2020, but I ran into the commission issue: it's hard to resolve this in a non-custodial plugin (i.e. without ANY custodial element in the middle.) If you want to create another custodial solution then this is not an issue of course: all you'd need is licenses.

**Reply by qlpqlp** [reply]
Can you explain more about the " commission issue" 

Wen I started last year I saw this docs:

https://github.com/Shopify/shopify-api-php
https://shopify.dev/docs/apps/payments/getting-started
https://shopify.dev/docs/api/functions

Then I realized that I had to code a SaaS and also have a minimum of XXX in Shopify sells for the Payment Gateway to be listed there, so I give up at that time, and I shared my process to implement here https://twitter.com/inevitable360/status/1574111231719477249 

:)

**Comment by georgeartem** [comment]
Happy to hear it! Shopify recently allowed third party payment providers on
their platform and based on what I am hearing, there are a few ways to get
it done. The team I am working with is more versed on this. The biggest
hurdle I am running into at the moment is finding a way to get them paid 😓
https://help.shopify.com/en/manual/payments/third-party-providers

**Reply by patricklodder** [reply]
You could transparently add fee outputs using [PSBT](https://github.com/bitcoin/bips/blob/master/bip-0174.mediawiki) instead of the currently prevalent [URI-based QR codes](https://github.com/bitcoin/bips/blob/master/bip-0021.mediawiki) but the wallet ecosystem needs a reason to implement this. Some wallets do support PSBT, many support WalletConnect. What is not supported is PSBT over WalletConnect... yet.

**Comment by georgeartem** [comment]
Sounds like "commission issue" is related to the business model of payment
processing, which is to take a small commission as the gateway that
validates the transaction. Please correct me here if I have it wrong.

PSBT is a neat concept. Has there been any effort to test scalability?

So far, we've been approaching this from the custodial (escrow) side as we
thought that we could reach into supporting use cases for escrow based
marketplaces like upwork or even crowdfunding apps like go or "doge"fund
me. . .

**Reply by patricklodder** [reply]
Don't worry about scaling. PSBT is an encoding standard, it doesn't "scale" or "not scale"... it just takes some (minimal) processing power to do the encoding. But it's all binary so parsing a JSON message is more expensive than that.

The custodial route you describe sounds like what coin payments did/does. You'll need some licenses for it because you'll be in (full, or at least can veto) control of the money at some points in the chain.

**Comment by georgeartem** [comment]
What are the licenses and how to go about getting them?

So what I am getting is that PSBT is a type of middle ware for applications?

Would it be possible to implement it for those running doge nodes and
dogecoin merchant apps like the type proposed here? Thereby by reducing the
need for wallet implementations?

**Reply by patricklodder** [reply]
> What are the licenses and how to go about getting them?

I cannot advise you on licenses - other than mentioning that you'll need them - because I'm not a lawyer, they change all the time in the different jurisdictions, and my 2020 information is definitely outdated. I'd strongly recommend to seek legal advice on that from a qualified firm if you're looking to go the custodial route.

> PSBT is a type of middle ware for applications?

It's an information exchange standard for unfinished transactions (which is basically what a payment request is.)

> Would it be possible to implement it for those running doge nodes and dogecoin merchant apps like the type proposed here?

Yes. It is available in the (unfinished) 1.21-dev branch for Dogecoin Core, but that is not a good functional wallet for day-to-day use, and the only interface available is to copy & paste the psbt data.

It is needed to define an implementation (ideally on top of https://github.com/ChainAgnostic and/or https://github.com/WalletConnect) because that makes it multi-chain capable - recent discussions I've had with Bitcoin and Litecoin developers show that there is a desire/need there too.

> Thereby by reducing the need for wallet implementations?

Would just start with a pilot with 1 or 2 wallets. I'm sure there are cool wallet providers out there that would love to work on this.

**Comment by georgeartem** [comment]
Re: licensing - you must be referring to "money transmitter licenses" that
are available through state regulating agencies that are common for payment
processors. Correct?
